### PR TITLE
SDLC-1411 Iseyer/fix dm issue

### PR DESCRIPTION
**Pull request by ianseyer**
Replaces inappropriate usage of lstrip with a removeprefix function to safely remove `spaces` without side effects

### CODE REVIEW

**Review by manatarms — APPROVED** [review]

**Comment by ianseyer** [comment]
testing

File: hangouts_chat.py [hunk 1]
```diff
@@ -99,7 +103,7 @@ def get_spaces(self) -> Iterable[dict]:
         return self._list('spaces', 'spaces')
 
     def get_space(self, name: str) -> Optional[dict]:
-        return self._request('spaces/{}'.format(name.lstrip('spaces/')))
+        return self._request('spaces/{}'.format(removeprefix(name, 'spaces/')))
 
     def get_members(self, space_name: str) -> Iterable[dict]:
         return self._list('spaces/{}/members'.format(space_name.lstrip('spaces/')), 'memberships')
```
Comment (porty) [inline hangouts_chat.py:109]: 👀

File: hangouts_chat.py [hunk 2]
```diff
@@ -15,6 +15,10 @@
 
 log = logging.getLogger('errbot.backends.hangoutschat')
 
+def removeprefix(source, prefix):
+    if source.startswith(prefix):
+        return source[len(prefix):]
+    return source[:]
```
Comment (sinkingpoint) [inline hangouts_chat.py:21]: Why `source[:]` over `source`?

Comment (sinkingpoint) [reply hangouts_chat.py:21]: Can we add type definitions here as well?

**Review by porty — APPROVED** [review]

**Review by larcher — APPROVED** [review]

**Pull request merged**